.github/workflows/maven-publish.yaml

```diff
@@ -41,14 +44,66 @@ jobs:
           git config --global user.email "envoy-bot@users.noreply.github.com"
           git config --global user.name "envoy-bot"
 
-      - name: Publish to Apache Maven Central
+      - name: Install with Maven
         working-directory: ${{ github.workspace }}/java
+        run: mvn -B install
+
+      - name: 'Configure gpg signing'
+        env:
+          GPG_KEY: ${{ secrets.gpg_key }}
+          GPG_KEY_NAME: ${{ secrets.gpg_key_name }}
+          GPG_PASSPHRASE: ${{ secrets.gpg_passphrase }}
         run: |
-          mvn release:prepare --batch-mode \
-          -DreleaseVersion=${{ github.ref_name }} \
-          -DdevelopmentVersion=${{ github.ref_name }}-SNAPSHOT \
-          -Dtag=${{ github.ref_name }} -X
+          # https://github.com/keybase/keybase-issues/issues/2798
+          export GPG_TTY=$(tty)
+          # Import gpg keys and warm the passphrase to avoid the gpg
+          # passphrase prompt when initating a deploy
+          # `--pinentry-mode=loopback` could be needed to ensure we
+          # suppress the gpg prompt
+          echo $GPG_KEY | base64 --decode > signing-key
+          gpg --passphrase $GPG_PASSPHRASE --batch --import signing-key
+          shred signing-key
+          
+          find . -type f \( -iname "pom.xml" -o -iname "*.jar" \) -print0 | while read -d $'\0' file
+          do
+            gpg --pinentry-mode=loopback --passphrase $GPG_PASSPHRASE -ab $file
+          done
+
+      - name: 'Release to sonatype repository'
+        working-directory: ${{ github.workspace }}/java
         env:
-          MAVEN_USERNAME: ${{ env.SONATYPE_USER }}
-          MAVEN_CENTRAL_TOKEN: ${{ env.SONATYPE_PASSWORD }}
-          MAVEN_GPG_PASSPHRASE: ${{ env.GPG_PASSPHRASE }}
+          READWRITE_USER: ${{ secrets.SONATYPE_USER }}
+          READWRITE_API_KEY: ${{ secrets.SONATYPE_PASSWORD }}
+          ENVOY_PROXY_PROFILE_ID: ${{ secrets.ENVOY_PROXY_PROFILE_ID }}
+        run: |
+          python sonatype_nexus_upload.py \
```

Comment:
This publishes then in the right artifact directory?

Like `pgv-java-grpc-0.6.9-SNAPSHOT` goes to the directory `~/.m2/repository/io/envoyproxy/protoc-gen-validate/pgv-java-grpc/0.6.8-SNAPSHOT/pgv-java-grpc-0.6.8-SNAPSHOT.jar` when you use the `--local` flag for the script?

Reply:
these were retrieved by running this on the runner.
```
find . -type f \( -iname "pom.xml" -o -iname "*.jar" \) -exec printf "%s\n" {}" \ " \;
```
you can see the successful run [here](https://github.com/envoyproxy/protoc-gen-validate/actions/runs/3137009907/jobs/5094640751#step:7:50)